packages/devextreme/testing/tests/DevExpress.ui.widgets/multiView.tests.js

```diff
@@ -1092,6 +1222,104 @@ QUnit.module('keyboard navigation', {
         delete this.animationStartAction;
     }
 }, () => {
+    QUnit.test('selectedIndex should be correct when one item is not visible using right arrow key', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [
+                { text: '1', visible: true },
+                { text: '2', visible: false },
+                { text: '3', visible: true }
+            ],
+            focusStateEnabled: true
+        });
+        const instance = $multiView.dxMultiView('instance');
+        const $lastItem = $multiView.find(`.${MULTIVIEW_ITEM_CLASS}`).eq(2);
+
+        $multiView.focusin();
+        keyboardMock($multiView).keyDown('right');
+
+        assert.strictEqual(instance.option('selectedIndex'), 2, 'correct index is selected');
+        assert.roughEqual(position($multiView), position($lastItem), 1, 'container moved to proper position');
+    });
+
+    QUnit.test('selectedIndex should be correct when one item is not visible using left arrow key', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [
+                { text: '1', visible: true },
+                { text: '2', visible: false },
+                { text: '3', visible: true }
+            ],
+            selectedIndex: 2,
+            focusStateEnabled: true
+        });
+        const instance = $multiView.dxMultiView('instance');
+        const $firstItem = $multiView.find(`.${MULTIVIEW_ITEM_CLASS}`).eq(0);
+
+        $multiView.focusin();
+        keyboardMock($multiView).keyDown('left');
+
+        assert.strictEqual(instance.option('selectedIndex'), 0, 'correct index is selected');
+        assert.roughEqual(position($multiView), -position($firstItem), 1, 'container moved to proper position');
+    });
+
+    QUnit.test('when only one item is visible, does not move the current visible item using all supported kbn keys', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [
+                { text: '1', visible: true },
+                { text: '2', visible: false }
+            ],
+            loop: true,
+            focusStateEnabled: true
+        });
+        const instance = $multiView.dxMultiView('instance');
+        const $itemContainer = $multiView.find(toSelector(MULTIVIEW_ITEM_CONTAINER_CLASS));
+
+        $multiView.focusin();
+        keyboardMock($multiView).keyDown('left');
```

Comment:
[copypaste]
We don't need creating keyboardMock every time we want to emulate key pressing
```suggestion
        const keyboard = keyboardMock($multiView);
        keyboard.keyDown('left');
```

Also u can use `forEach` to reduce copypaste while all asserts are the same

Reply:
[6ecc98e](https://github.com/DevExpress/DevExtreme/pull/28137/commits/6ecc98e35f609c4eb9d7d8b7c3b7f68d63eb3670)